interpreter/prune.go

```diff
@@ -298,3 +381,14 @@ func (p *astPruner) existsWithKnownValue(id int64) bool {
 	val, valueExists := p.value(id)
 	return valueExists && !types.IsUnknown(val)
 }
+
+func (p *astPruner) nextID() int64 {
+	_, found := p.state.Value(p.nextExprID)
+	for found {
+		p.nextExprID++
+		_, found = p.state.Value(p.nextExprID)
+	}
```

Comment:
Consider, even if only briefly, the following idiom (which makes up for the [control structure that's missing](https://en.wikipedia.org/wiki/Control_flow#Loop_with_test_in_the_middle) in most languages):

```
for true {
        _, found := p.state.Value(p.nextExprID)
        if found {
                break
        }
        p.nextExprID++
}
```